ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +360,709 @@ const createSecret = async (res, appid, payload, DB) => {
+            limit: 1,
+        }
+        let secret_metadata_result = await DB.partitionedFind('secret_metadata', query)
+
+        // build result
+        let result = {}
+        if (secret_metadata_result.docs.length > 0) {
+            const doc = secret_metadata_result.docs[0]
+            result['secretName'] = base64_decode(doc['secretName'])
+            result['description'] = base64_decode(doc['description'])
+            result['createTime'] = doc['createTime']
+            if (doc['plannedDeleteTime']) {
+                result['plannedDeleteTime'] = doc['plannedDeleteTime']
+            }
+            if (doc['rotationInterval']) {
+                result['rotationInterval'] = doc['rotationInterval']
+                result['lastRotationDate'] = doc['lastRotationDate']
+                result['nextRotationDate'] = doc['nextRotationDate']
+            }
+        }
+
+        res.send(_result(200, 'describe secrets success.', result))
+        return
+    } catch (e) {
+        console.info('describeSecrets :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * Force delete secret or schedule a time to delete secret
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}secretName(String [1~64]) : The name of the secret. eg. 'secretName01'
+ *          ==> {o}recoveryPeriod(Number [1~365])[defalut='30']: Specifies the recovery period of the secret,The unit is day, if you do not forcibly delete it.
+ *               eg. '50' 
+ *          ==> {o}forceDelete(String [true/false])[defalut='false']: Specifies whether to forcibly delete the secret.
+ *                 If this parameter is set to true, the secret cannot be recovered. eg. 'true'
+ * @returns
+ */
+const deleteSecret = async (res, appid, DB, payload) => {
+    try {
+        //get and check Param in payload
+        const deleteTime = new Date().getTime()
+        let secretName = getParam_String(payload, 'secretName')
+        let recoveryPeriod = payload['recoveryPeriod']
+        let forceDelete = payload['forceDelete']
+        if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+        if (typeof (recoveryPeriod) !== 'number' && recoveryPeriod != '' && recoveryPeriod != undefined) {
+            res.send(_result(400, 'recoveryPeriod must be number'))
+            return
+        }
+        if (recoveryPeriod < 1 || recoveryPeriod > 365) {
+            res.send(_result(400, 'recoveryPeriod must be more then 0 and less than 366'))
+            return
+        }
+        if (forceDelete != '' && forceDelete != undefined) {
+            if (forceDelete != 'true' && forceDelete != 'false') {
```

Comment:
TRUE/FALSE, True/False is also acceptable.

Reply:
will fix